src/inc/msquic.h

```diff
@@ -875,6 +875,23 @@ typedef struct QUIC_STREAM_STATISTICS {
     uint64_t StreamBlockedByAppUs;
 } QUIC_STREAM_STATISTICS;
 
+typedef enum QUIC_AEAD_ALGORITHM_TYPE {
+    QUIC_AEAD_ALGORITHM_AES_128_GCM = 0,
+    QUIC_AEAD_ALGORITHM_AES_256_GCM = 1,
+    QUIC_AEAD_ALGORITHM_CHACHA20_POLY1305 = 2,
+} QUIC_AEAD_ALGORITHM_TYPE;
+
+#pragma warning(push)
+#pragma warning(disable:4200)  // nonstandard extension used: zero-sized array in struct/union
+typedef struct QUIC_STATELESS_RETRY_CONFIG {
+    QUIC_AEAD_ALGORITHM_TYPE Algorithm; // AEAD algorithm for the key.
+    uint32_t RotationMs;                // Key rotation interval in milliseconds.
+    uint32_t SecretLength;              // Length of the secret.
+    _Field_size_bytes_(SecretLength)
+        uint8_t Secret[0];              // Secret to generate the key.
```

Comment:
I think this is the first zero-sized array we introduce in the API, which is significant (zero-sized array are error prone).
Other structure seems to be using a pointer to memory the user can put wherever they see fit (`QUIC_SCHANNEL_CREDENTIAL_ATTRIBUTE_W`).

We should stay consistent.

Reply:
I prefer your suggestion as well.  This being the first zero-length array in the header has caused some annoyance getting the compiler errors ignored.

Reply:
After the fact, I found one case: `uint16_t ProcessorList[1];`.

I still don't think we should introduce more of it (I'd rather consider fixing that one since it is still preview).
With a pointer, the user still can allocate everything in one memory block if they want to, and it is less error prone.

Reply:
The version settings use pointers and a count.  Given the push for standardization, I think that we may find more hurdles to zero-length arrays in the header (if not also the internal uses of it) in the future.